wordpress-mobile/WordPress-Android
# Issue/13328 restore skeleton

Parent #13328 

This PR introduces the "Restore" feature as accessed via the activity log list. This is prep PR.
 
`RestoreActivity`, `RestoreViewModel` are copies of the `BackupDownloadActivity` and `BackupDownloadViewModel` and functionality for observables and the wizard remain the same. Differences include supporting class renames, as well as function name changes. I have left //todo's to fill in particulars for each step as I include them.

`RestoreDetailsFragment` and `RestoreDetailsViewModel` are mere skeletons because it would have been too much to review. 

I did not use the `Rewind` package, but created a new one for `Restore`. I didn't rename because the current classes in `Rewind` are used by the ActivityLog. These `rewind` classes may be removed in the future and if so, this package will be deleted in its entirety.

This PR is a bit larger than normal, but to show that it works I had to go all the way to presenting the details view. It's best reviewed by commits which I have broken down as follows:

- Dagger & App Housekeeping: 9e4ad1b 
- Feature Flag in ActivityLog: ad0cc55
-- Adds support for showing the new restore wizard when the `RestoreFeatureConfig` is enabled. This will work with or without the BackupFeatureConfig flag.
- Restore Activity/Details: 723852c
-- This includes the cloned classes from BackupDownload feature
- Skeleton Unit Tests: 4f9d98b

**Merge Instructions**
- <del>Make sure #13759 has been merged<del> ✅
- <del>Remove Not Ready For Merge Label<del> ✅
- Merge as normal

**To test:**
- Launch the app with RestoreFeatureConfig flag on and the BackupFeatureFlag off
- Navigate to ActivityLog
- Tap the rewind icon for an item
- Note that an empty screen with the "Restore" title is shown

- Launch the app with RestoreFeatureConfig flag on and the BackupFeatureFlag on
- Navigate to ActivityLog
- Tap the More menu
- Tap Restore to this point menu item
- Note that an empty screen with the "Restore" title is shown

- Launch the app with RestoreFeatureConfig flag off and the BackupFeatureFlag off
- Navigate to ActivityLog
- Tap the rewind icon for an item
- Note that the restore dialog box is shown

PR submission checklist:
